### PR TITLE
feat(EPS): add eps enterprise projects data source

### PR DESCRIPTION
**Pull request by luoping-12345**
**What this PR does / why we need it**:
 add eps enterprise projects data source

**Which issue this PR fixes**:

**Special notes for your reviewer**:

**Release note**:

```
add eps enterprise projects data source
```
```
./scripts/codecheck.sh ./huaweicloud/services/eps

==> Checking for running environment...

==> Applying patch...

==> Checking for code complexity...
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Language                              Files     Lines   Blanks  Comments     Code Complexity Complexity/Lines
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Go                                        3       468       66         3      399         51            37.70
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
~resource_huaweicloud_enterprise_project.go       215       36         1      178         27            15.17
~a_source_huaweicloud_enterprise_project.go       115       14         2       99         15            15.15
~_source_huaweicloud_enterprise_projects.go       138       16         0      122          9             7.38
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Total                                     3       468       66         3      399         51            37.70
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Processed 12418 bytes, 0.012 megabytes (SI)
─────────────────────────────────────────────────────────────────────────────────────────────────────────────

the TOP10 most complex functions:
7 eps resourceEnterpriseProjectUpdate huaweicloud/services/eps/resource_huaweicloud_enterprise_project.go:144:1
6 eps dataSourceEnterpriseProjectsRead huaweicloud/services/eps/data_source_huaweicloud_enterprise_projects.go:96:1
6 eps flattenEnterpriseProject huaweicloud/services/eps/data_source_huaweicloud_enterprise_project.go:90:1
5 eps resourceEnterpriseProjectRead huaweicloud/services/eps/resource_huaweicloud_enterprise_project.go:109:1
5 eps dataSourceEnterpriseProjectRead huaweicloud/services/eps/data_source_huaweicloud_enterprise_project.go:51:1
4 eps resourceEnterpriseProjectDelete huaweicloud/services/eps/resource_huaweicloud_enterprise_project.go:183:1
3 eps resourceEnterpriseProjectCreate huaweicloud/services/eps/resource_huaweicloud_enterprise_project.go:84:1
1 eps ResourceEnterpriseProject huaweicloud/services/eps/resource_huaweicloud_enterprise_project.go:20:1
1 eps flattenEnterpriseProjects huaweicloud/services/eps/data_source_huaweicloud_enterprise_projects.go:82:1
1 eps DataSourceEnterpriseProjects huaweicloud/services/eps/data_source_huaweicloud_enterprise_projects.go:17:1
Average: 3.64

==> Checking for golangci-lint...

==> Checking for Nolint directives...

==> Checking for TF features in eps...

==> Checking for misspell in eps...

==> Checking for code complexity in ./huaweicloud/services/acceptance/eps...
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Language                              Files     Lines   Blanks  Comments     Code Complexity Complexity/Lines
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Go                                        3       232       28         0      204         11            11.83
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
~rce_huaweicloud_enterprise_project_test.go       108       15         0       93         11            11.83
~ce_huaweicloud_enterprise_projects_test.go        89        7         0       82          0             0.00
~rce_huaweicloud_enterprise_project_test.go        35        6         0       29          0             0.00
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Total                                     3       232       28         0      204         11            11.83
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Processed 7445 bytes, 0.007 megabytes (SI)
─────────────────────────────────────────────────────────────────────────────────────────────────────────────

the TOP5 most complex functions:
6 eps testAccCheckEnterpriseProjectDestroy huaweicloud/services/acceptance/eps/resource_huaweicloud_enterprise_project_test.go:71:1
2 eps getResourceEnterpriseProject huaweicloud/services/acceptance/eps/resource_huaweicloud_enterprise_project_test.go:16:1
1 eps testAccEnterpriseProject_update huaweicloud/services/acceptance/eps/resource_huaweicloud_enterprise_project_test.go:102:1
1 eps testAccEnterpriseProject_basic huaweicloud/services/acceptance/eps/resource_huaweicloud_enterprise_project_test.go:94:1
1 eps TestAccEnterpriseProject_basic huaweicloud/services/acceptance/eps/resource_huaweicloud_enterprise_project_test.go:25:1
Average: 1.75

==> Checking for golangci-lint in ./huaweicloud/services/acceptance/eps...

==> Checking for Nolint directives in ./huaweicloud/services/acceptance/eps...

==> Cleanup patch...

Check Completed!
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

SDK:  add query params.  Verify whether related functions are normal.
```
make testacc TEST="./huaweicloud/services/acceptance/eps" TESTARGS="-run TestAccEnterpriseProjectDataSource_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/eps -v -run TestAccEnterpriseProjectDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccEnterpriseProjectDataSource_basic
=== PAUSE TestAccEnterpriseProjectDataSource_basic
=== CONT  TestAccEnterpriseProjectDataSource_basic
--- PASS: TestAccEnterpriseProjectDataSource_basic (11.32s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/eps       11.378s
```
```
make testacc TEST="./huaweicloud/services/acceptance/eps" TESTARGS="-run TestAccEnterpriseProjectsDataSource_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/eps -v -run TestAccEnterpriseProjectsDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccEnterpriseProjectsDataSource_basic
=== PAUSE TestAccEnterpriseProjectsDataSource_basic
=== CONT  TestAccEnterpriseProjectsDataSource_basic
--- PASS: TestAccEnterpriseProjectsDataSource_basic (34.80s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/eps       34.842s
```

### CODE REVIEW

File: docs/data-sources/eps_enterprise_projects.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+The following arguments are supported:
+
+* `name` - (Optional, String) Specifies the enterprise project name. Fuzzy search is supported.
+
+* `enterprise_project_id` - (Optional, String) Specifies the ID of an enterprise project.
+  The value **0** indicates enterprise project default.
+
+* `status` - (Optional, Int) Specifies the status of an enterprise project. The valid values are as follows:
+  + **1** indicates Enabled.
+  + **2** indicates Disabled.
```
Comment (jinyangyang222) [inline docs/data-sources/eps_enterprise_projects.md:30]: + **1**: Enabled.
  + **2**: Disabled.

File: docs/data-sources/eps_enterprise_projects.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+* `type` - (Optional, String) Specifies the type of an enterprise project. The valid values are as follows:
+  + **prod** indicates Commercial project.
+  + **poc** indicates Test project.
+
```
Comment (jinyangyang222) [inline docs/data-sources/eps_enterprise_projects.md:35]: + **prod**: Commercial project.
  + **poc**: Test project.

File: docs/data-sources/eps_enterprise_projects.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - Data source ID.
+
+* `enterprise_projects` - All groups that match the filter parameters.
+  The [enterprise_projects](#Enterprise_Projects) structure is documented below.
+
+<a name="Enterprise_Projects"></a>
+The `groups` block supports:
```
Comment (jinyangyang222) [inline docs/data-sources/eps_enterprise_projects.md:46]: The `enterprise_projects` block supports:

File: docs/data-sources/eps_enterprise_projects.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  + **prod** indicates Commercial project.
+  + **poc** indicates Test project.
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - Data source ID.
+
+* `enterprise_projects` - All groups that match the filter parameters.
```
Comment (jinyangyang222) [inline docs/data-sources/eps_enterprise_projects.md:42]: * `enterprise_projects` - All projects that match the filter parameters.

File: docs/data-sources/eps_enterprise_projects.md (hunk 1 above)
Comment (luoping-12345) [reply docs/data-sources/eps_enterprise_projects.md]: fixed

File: docs/data-sources/eps_enterprise_projects.md (hunk 2 above)
Comment (luoping-12345) [reply docs/data-sources/eps_enterprise_projects.md:35]: fixed

File: docs/data-sources/eps_enterprise_projects.md (hunk 3 above)
Comment (luoping-12345) [reply docs/data-sources/eps_enterprise_projects.md]: fixed

File: docs/data-sources/eps_enterprise_projects.md (hunk 4 above)
Comment (luoping-12345) [reply docs/data-sources/eps_enterprise_projects.md]: fixed

File: huaweicloud/services/acceptance/eps/data_source_huaweicloud_enterprise_projects_test.go [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+data "huaweicloud_enterprise_projects" "test" {}
+
+locals {
+	enterprise_project_id = data.huaweicloud_enterprise_projects.test.enterprise_projects[0].id
+}
+data "huaweicloud_enterprise_projects" "enterprise_project_id_filter" {
+	enterprise_project_id = local.enterprise_project_id
+}
+output "enterprise_project_id_filter_is_useful" {
+	value = length(data.huaweicloud_enterprise_projects.enterprise_project_id_filter.enterprise_projects) > 0 && alltrue(
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/eps/data_source_huaweicloud_enterprise_projects_test.go]: please use two spaces as the indent.

Comment (luoping-12345) [reply huaweicloud/services/acceptance/eps/data_source_huaweicloud_enterprise_projects_test.go]: fixed

**Pull request merged**